packages/common/src/search/_internal/commonHelpers/getNextPortalCallback.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  ArcGISIdentityManager,
+  IRequestOptions,
+} from "@esri/arcgis-rest-request";
+import { cloneObject } from "../../../util";
+import { IHubSearchResponse } from "../../types/IHubSearchResponse";
+import { IPagingParams } from "@esri/arcgis-rest-portal";
+
+type ICommonNextOptions = IRequestOptions & IPagingParams;
+
+/**
+ * @private
+ * Create a `.next()` function for portal search type
+ * @param request
+ * @param nextStart
+ * @param total
+ * @param fn
+ * @returns
+ */
+export function getNextPortalCallback<I extends ICommonNextOptions, O>(
```

Comment:
This is a more streamlined version of the `getNextFunction()` helper that was mistakenly exposed to the public. We no longer have to check for `options.requestOptions` as rest.js 4x spread them onto the top-level object.

NOTE: I have to use the double generic syntax since this function is used for 4 different portal searches, making type compatibility a nightmare.